### PR TITLE
Newbranch

### PR DESCRIPTION
**Pull request by SomyaGandhi**
Fixes issue #2170 

#### Checklist:
- [ ] 4 space indentation.
- [ ] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [ ] Sample Input / Output is added at the end of file.
- [ ] Logic Documentation (Comments).
- [ ] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- 
- 

#### Files Added:
- ReadMe file
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Quicksort_3_way/README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+3-Way Quicksort
```
Comment (raghav-dalmia) [inline Quicksort_3_way/README.md:1]: use # before this line

File: Quicksort_3_way/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+3-Way Quicksort
+
+
+What is Quicksort?
```
Comment (raghav-dalmia) [inline Quicksort_3_way/README.md:4]: use ## before this

File: Quicksort_3_way/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+3-Way Quicksort
+
+
+What is Quicksort?
+Quicksort is a method of sorting in which a pivot element is picked and the array is partioned according to this element.It is a divide and conquer algorithm. It divides the array in two parts, the left of pivot element has elements lesser than the pivot element and the right side has elements greater than the pivot element.
```
Comment (raghav-dalmia) [inline Quicksort_3_way/README.md:5]: Leave an empty line before this

File: Quicksort_3_way/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+3-Way Quicksort
+
+
+What is Quicksort?
+Quicksort is a method of sorting in which a pivot element is picked and the array is partioned according to this element.It is a divide and conquer algorithm. It divides the array in two parts, the left of pivot element has elements lesser than the pivot element and the right side has elements greater than the pivot element.
+
+
+What is 3-Way Quicksort?
```
Comment (raghav-dalmia) [inline Quicksort_3_way/README.md:8]: use ## before this line

File: Quicksort_3_way/README.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+
+What is 3-Way Quicksort?
+3-Way quicksort is nearly same as quicksort. The only difference is that the elements are divided into three parts:-
+1. Less than pivot element
+2. Greater than pivot element
+3. Equal to pivot element
+The idea of 3 way QuickSort is to process all occurrences of pivot and is based on Dutch National Flag algorithm.
+
+Example:-
```
Comment (raghav-dalmia) [inline Quicksort_3_way/README.md:15]: Bold this line

File: Quicksort_3_way/README.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+If the given array is {1, 4, 2, 4, 2, 4, 1, 2, 4, 1, 2, 2, 2, 2, 4, 1, 4, 4, 4 ,5} Let's take the pivot element as 4. Then the array will be partioned into 3 parts i.e- less than 4 (1,2,2,1,2,1,2,2,2,2,1), equal to 4 (4,4,4,4,4,4,4,4) and greater than 4 (5).
+
+
+Pseudocode
```
Comment (raghav-dalmia) [inline Quicksort_3_way/README.md:19]: use ##

File: Quicksort_3_way/README.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    for(int k = right - 1; k > q; k--, i++) //Bringing pivot element and its duplicates to the correct position
+        swap(array[i], array[k]);
+
+    Quick_Sort_3way(array, left, j); //Sorting elements on the left of pivot
+    Quick_Sort_3way(array, i, right); //Sorting elements on the right of pivot
+
+Refer:- https://learnforeverlearn.com/pivot_3way_music/  for explanation
+
+
+Properties of 3-Way Quicksort
```
Comment (raghav-dalmia) [inline Quicksort_3_way/README.md:65]: use ##

File: Quicksort_3_way/README.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+It is not stable.
+It uses O(log(n)) extra space because of the recursion.
+Worst case run time is as same as classic quicksort, O(n2), but typically O(nlog(n)) time
+
+
+Time Complexity:- O(n^2)
```
Comment (raghav-dalmia) [inline Quicksort_3_way/README.md:72]: use ##

**Comment by raghav-dalmia** [comment]
Make sure to squash your commits after these changes.

**Comment by MastersAbh** [comment]
Hi @SomyaGandhi I see no update for over 20 days, so closing it for now. Reach out if you still want to work on this.

**Pull request closed without merging**